engines/osxnative/sandbox.go

```diff
@@ -136,21 +125,99 @@ func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
 	cmd := exec.Command(s.taskPayload.Command[0], s.taskPayload.Command[1:]...)
 	cmd.Stdout = stdoutLogWriter{s.context}
 	cmd.Stderr = stderrLogWriter{s.context}
+
+	// USER and HOME are treated seperately because their values
+	// depend on either we create the new user successfuly or not
+	processUser := os.Getenv("USER")
+	processHome := os.Getenv("HOME")
+
+	// If we fail to create a new user, the most probable cause is that
+	// we don't have enough permissions. Chances are that we are running
+	// in in a development environment, so do not fail the task to tests
+	// run successfully.
+	u := user{}
+	if err = u.create(); err != nil {
```

Comment:
So I think that this can be rather dangerous.  If for some reason creating a user fails, this could be run under escalated permissions without knowing it.

I think it should be an either/or thing.  Either the engine is configured to run a process under a temp user and fails if that user can't be used, or the engine is configured to not create a temp user and run under the worker user.